app/utils/ImplicitsWithFunUtils.scala

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import play.api.mvc.Result
+
+import scala.concurrent.Future
+import scala.language.implicitConversions
+
+object ImplicitsWithFunUtils {
```

Comment:
this should come via proposal stage as this fundamentals different from the way we do in the rest of the code base; the short answer, not in the scope of this PR pls

Reply:
okay no problem, will move code to controller